src/Geometry.cpp

```diff
@@ -1,10 +1,19 @@
 #include "Geometry.h"
 
-Geometry::Geometry(std::vector<glm::vec2>& v, std::vector<glm::vec3>& c)
+Geometry::Geometry(const Geometry& g)
+  : vertexBuffer(g.vertexBuffer)
+  , colourBuffer(g.colourBuffer)
+  , vertexArray(g.vertexArray)
+  , vertices(g.vertices)
+  , colours(g.colours)
+{
+}
+
+Geometry::Geometry(std::vector<glm::vec2> v, std::vector<glm::vec3> c)
   : vertexBuffer(0)
   , colourBuffer(0)
   , vertexArray(0)
-  , vertices(v)
-  , colours(c)
+  , vertices(std::move(v))
+  , colours(std::move(c))
```

Comment:
This makes use of move semantics. Starting c++11, move semantics allow essentially moving data without copying it. Downside to this is that whatever "v" and "c" referred to originally will not be valid.

I find this to be an acceptable trade off because we shouldn't need the original copy of the vertex and colour data after initializing the geometry.